public/app/core/components/search/search.ts

```diff
@@ -2,12 +2,32 @@ import _ from 'lodash';
 import coreModule from '../../core_module';
 import { SearchSrv } from 'app/core/services/search_srv';
 import { contextSrv } from 'app/core/services/context_srv';
+
 import appEvents from 'app/core/app_events';
+import { parse, SearchParserOptions, SearchParserResult } from 'search-query-parser';
+import { getDashboardSrv } from 'app/features/dashboard/services/DashboardSrv';
+export interface SearchQuery {
+  query: string;
+  parsedQuery: string | SearchParserResult;
```

Comment:
why string or SearchParserResult?

Reply:
It's specific to how search-query-parser works. If the query does not contain any keywords, then it will return a string. Gonna unify that to be SearchParserResult only